dimod/binary/binary_quadratic_model.py

```diff
@@ -758,6 +758,93 @@ def add_linear_equality_constraint(
                     u, v, 2 * lagrange_multiplier * ubias * vbias)
         self.offset += lagrange_multiplier * constant * constant
 
+    def add_linear_inequality_constraint(
+            self, terms: Iterable[Tuple[Variable, Bias]],
+            lagrange_multiplier: Bias,
+            label: str,
+            constant: int = 0,
+            lb: int = - np.Infinity,
+            ub: int = 0,
+            slack_method: str = "log2",
+            cross_zero: bool = False
+    ):
+        """Add a linear inequality constraint as a quadratic objective.
+
+        Adds a linear inequality constraint of the form:
+
+        math:'lb <= \sum_{i,k} a_{i,k} x_{i,k} + constant <= ub'
+        to the binary quadratic model as a quadratic objective.
+        For constraints with fractional coefficients, multiply both sides of the inequality by an
+        appropriate factor of ten to attain or approximate integer coefficients.
+
+        Args:
+            terms (iterable/iterator):
+                An iterable of 2-tuples, (variable, bias).
+                Each tuple is evaluated to the term (bias * variable).
+                All terms in the list are summed.
+            lagrange_multiplier:
+                The coefficient or the penalty strength. This value is
+                multiplied by the entire constraint objective and added to the
+                bqm (it doesn't appear explicitly in the equation above).
+            label:
+                Prefix used to label the slack variables used to create the new objective.
+            constant:
+                The constant value of the constraint.
+            lb:
+                lower bound for the constraint
+            ub:
+                upper bound for the constraint
+            slack_method:
+                "The method for adding slack variables. Supported methods are:
+                - log2: Adds log2(slack_range) number of binary variables to the constraint.
+            cross_zero
+                If True, includes zero as the domain of constraint if not include already
+
```

Comment:
Need to document the return value, and add it to the function signature as a type annotation.